app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -25,57 +25,17 @@ class ManageIQ::Providers::Redhat::InfraManager < ManageIQ::Providers::Ovirt::In
   require_nested  :DistributedVirtualSwitch
   include_concern :ApiIntegration
 
+  has_one :network_manager,
+          :foreign_key => :parent_ems_id,
+          :class_name  => "ManageIQ::Providers::Redhat::NetworkManager",
```

Comment:
NOTE by declaring the class_name on the association here we don't have to override the `ensure_managers` method which was identical except for `build_network_manager(:type => 'ManageIQ::Providers::Redhat::NetworkManager')`